pkg/controller/machineautoscaler/machinetarget.go

```diff
@@ -341,7 +341,12 @@ func checkScaleFromZeroAnnotations(annotations map[string]string) (map[string]st
 		}
 		_, err = annotationsutil.ParseMachineSetAnnotationKey(annotations, annotationsutil.GpuTypeKey)
 		if err != nil {
-			annotations = annotationsutil.SetGpuTypeAnnotation(annotations, annotationsutil.GpuNvidiaType)
+			//If there are no gpus, gputype needs to be an empty string
```

Comment:
we need to make sure that the type label is applied even when it did not exist previously. i think you need to double check the logic for how we enter these clauses.

we want to add the type label, and we currently gate this logic on whether it exists already.

Reply:
ok, so it seems that the `SetGpuTypeAnnotation` is the culprit here.